### PR TITLE
feat(lib.menubar): added workaround for "scrolling" large list

### PR DESCRIPTION
**Pull request by actionless**

### CODE REVIEW

File: lib/menubar/utils.lua.in [hunk 1]
```diff
@@ -209,6 +209,21 @@ function utils.parse_dir(dir)
     return programs
 end
 
+--- Slice a table to given range.
```
Comment (blueyed) [inline lib/menubar/utils.lua.in]: Yes, I think it's a good fit for `awful.util.table`.

File: lib/menubar/utils.lua.in [hunk 2]
```diff
@@ -209,6 +209,21 @@ function utils.parse_dir(dir)
...
+-- @param original_table Original table.
+-- @param range_start Range start.
+-- @param range_finish Range finish.
+-- @return A sliced table.
```
Comment (blueyed) [inline lib/menubar/utils.lua.in]: Use `@tparam` here, which also provides type information, e.g. `@tparam table original_table`.

File: lib/menubar/utils.lua.in [hunk 3]
```diff
@@ -209,6 +209,21 @@ function utils.parse_dir(dir)
...
+function utils.table_range(original_table, range_start, range_finish)
+    range_finish = range_finish or #original_table
+    local result = {}
+    for i=range_start,range_finish do
```
Comment (blueyed) [inline lib/menubar/utils.lua.in]: Please add spaces around operators (`=`, ..) and `,` etc.

**Comment by blueyed** [comment]
> so can we go forward with current workaround?

I think it's a good fix for now.  Thanks!
I've left some minor comments, and would merge it after they are addressed - or could do so while merging - just let me know then.

**Comment by actionless** [comment]
i fixed it

but looking on change/conversation again i think what later it will be better make it paginated to have more "expectable" behavior

**Comment by actionless** [comment]
i tried to use `textbox._layout:get_pixel_extents()` for computing text_width before rendering but it became even slower than before on the large lists

**Comment by actionless** [comment]
so i tried on my fork:
https://github.com/actionless/awesome_config/commit/7ef72b7d7f3d3e76842e9bbd75e57852e665d6d9

if it looks ok i can rebase it on current master

**Comment by actionless** [comment]
closing in prior to https://github.com/awesomeWM/awesome/pull/144

**Pull request closed without merging**